[Kim Junseo] iP

jskimdev opened the pull request:
### ByteTalker

ByteTalker manages your time by remembering things you need to do. It's,

- text-based
- easy to use
- SUPER FAST

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jskimdev committed:
Rename Chatbot

jskimdev committed:
Echo user input

jskimdev committed:
Store message

jskimdev committed:
Include status to task

jskimdev committed:
Add types to task

jskimdev committed:
Add error handling

jskimdev committed:
Add delete task feature

jskimdev committed:
Automate Text UI testing

jskimdev committed:
Rename the class to chatbot name

guanquann commented on src/main/java/ByteTalker.java:
```diff
@@ -0,0 +1,135 @@
+        }
+    }
+
+    public void markTask(String[] split_message) {
```
In Java, I believe the standard naming convention is camelCase. `split_message` should be `splitMessage` instead.

I notice the same issue in several other places as well

jskimdev replied:
Ah missed that. Noted. Thanks for informing it.

guanquann commented on src/main/java/ByteTalker.java:
```diff
@@ -0,0 +1,135 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class ByteTalker {
```
Javadocs not present in class and methods. 

It should look something like this:
```
/**
 * Returns lateral location of the specified position.
 * If the position is unset, NaN is returned.
 *
 * @param x X coordinate of position.
 * @param y Y coordinate of position.
 * @param zone Zone of position.
 * @return Lateral location.
 * @throws IllegalArgumentException  If zone is <= 0.
 */
public double computeLocation(double x, double y, int zone)
        throws IllegalArgumentException {
    //...
}
```

jskimdev replied:
Added last week after the feedback!

guanquann commented on src/main/java/ByteTalker.java:
```diff
@@ -0,0 +1,135 @@
+public class ByteTalker {
+    private ArrayList<Task> storage = new ArrayList<>();
+
+    public void returnList () {
```
```suggestion
    public void returnList() {
```

guanquann commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+    Event("E") {
+        @Override
+        public String toString() {
+            return "[" + getIcon() + "]" + "[" + getStatusIcon() + "] " + getTask() + " (from: " + getFrom() + " to:" + getTo() + ")";
```
Line length should be no longer than 120 chars. Consider using linebreak

jskimdev replied:
Noted, I will try to break the lines exceeding 120 chars

guanquann commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+        this.isDone = false;
+    }
+
+    public void setTask(String task) {
```
Good use of getters and setters method!

jskimdev replied:
Thanks!

guanquann commented on src/main/java/ByteTalker.java:
```diff
@@ -0,0 +1,135 @@
+    }
+
+    public static void main(String[] args) {
+        /*
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        */
```
This comment seems unnecessary to me. Consider removing them.

guanquann commented on src/main/java/ByteTalkerException.java:
```diff
@@ -0,0 +1,19 @@
+public class ByteTalkerException {
+    public static class TODONoTaskException extends Exception {
```
The naming convention should not be in uppercase. Only constants should have uppercase. 

```suggestion
    public static class TodoNoTaskException extends Exception {
```
Same issue for DEADLINEException

guanquann reviewed (commented):
It seems like you are a little behind schedule. You should have completed the taskings in Week 3 already. Most of the comments should be corrected if you had completed the taskings in week 3.

You can take a look at [conventions](https://se-education.org/guides/conventions/java/intermediate.html) for more details on good coding standards. All the best!

jskimdev committed:
Create classes for each task and correct some names to follow code standards

jskimdev committed:
Save task in the hard disk(./data/ByteTalker.txt) automatically whenever the task list changes

jskimdev committed:
Fix code error

jskimdev committed:
Merge branch 'branch-Level-7'
Save tasks to the hard disk method done

jskimdev committed:
Convert date and time to LocalDateTime type instead of String

jskimdev committed:
Merge branch 'branch-Level-8'
Convert date and time to LocalDateTime type instead of String

jskimdev committed:
Remove unused import

jskimdev committed:
Create separate classes for UI, Parser, and TaskList

jskimdev committed:
Merge branch 'branch-A-MoreOOP'
Create separate classes done for TaskList, Parser, and Ui

jskimdev committed:
Group classes into packages

jskimdev committed:
Merge branch 'add-gradle-support'
Merge add-gradle-support branch to use gradle

jskimdev committed:
Add gradle and transfer task methods to TaskList

jskimdev committed:
Add testcases for JUnit testing and add more methods for Parser

jskimdev committed:
Add JavaDoc for Ui, Task, TaskType, and Task

jskimdev committed:
Add JavaDoc for TaskList, Todo, Event, Deadline, Storage, and ByteTalker

jskimdev committed:
Merge branch 'A-JavaDoc'
Add JavaDoc to most of the classes and methods

billyhoce commented on src/main/java/bytetalker/exception/ByteTalkerException.java:
```diff
@@ -0,0 +1,39 @@
+        }
+    }
+
+    public static class DeadlineUnsupportedFormatException extends Exception {
```
Seems like there is a `DeadlineWrongFormatException` and a `DeadlineUnsupportedFormatException`, could you share the reason to separate the two? It sounds like either can be used in the situation of a different format than expected!

jskimdev replied:
At the time of coding it, I was thinking of different errors that could rise, but realised that both errors are eventually the same! I will try to combine both exceptions

billyhoce commented on src/main/java/bytetalker/parser/Parser.java:
```diff
@@ -0,0 +1,104 @@
+    }
+
+    public static ArrayList<String> parseDeadlineInput(String[] splitMessages) throws ByteTalkerException.DeadlineUnsupportedFormatException {
+        ArrayList<String> storeMessages = new ArrayList<>();
```
An ArrayList object here is named with a verb name `storeMessages`, maybe a more noun-like name like `messageContainer` or `messageStore` can be considered!

jskimdev replied:
Thank you for feedback! I will think about it

billyhoce commented on src/main/java/bytetalker/task/TaskType.java:
```diff
@@ -0,0 +1,34 @@
+    TODO("T") {
+    },
+    DEADLINE("D") {
+
```
Maybe you can standardise the formatting for the enum here. The first object "TODO" has no empty line after the curly bracket but "DEADLINE" and "EVENT" has one.

jskimdev replied:
Oh, alright I will try to standardise the format. Thank you for feedback!

billyhoce commented on src/main/java/bytetalker/ui/Ui.java:
```diff
@@ -0,0 +1,144 @@
+    /**
+     * Prints out error message if there is an error while changing the task list.
+     */
+    public void showErrorMsgStoreTasks() {
```
Maybe an alternate name to consider would be "showStoreTaskErrorMessage"

jskimdev replied:
That seems to make more sense. Thank you for feedback!

billyhoce reviewed (commented):
Overall well written code that is neat and easy to follow! Just a few very small fixes. Keep it up!

jskimdev committed:
Add data folder to gitignore

jskimdev committed:
Add find tasks by String method

jskimdev committed:
Merge branch 'branch-Level-9'
Add find tasks by String method

jskimdev committed:
Change naming for some methods to follow coding standard

jskimdev committed:
Merge branch 'branch-A-CodingStandard'
Change naming for some methods to follow coding standard

jskimdev committed:
Add GUI for the chatbot using JavaFx

jskimdev committed:
Add JavaFx using gradle dependency

jskimdev committed:
Merge branch 'branch-Level-10'
Add Gui to the chatbot via JavaFx

jskimdev committed:
Test on a new device

jskimdev committed:
Test on new device using git CLI

jskimdev committed:
Change main class name in build.gradle

jskimdev committed:
Add assertions to ensure assumptions true

jskimdev committed:
Extract some methods

There are several methods with long lines of codes and
can be extracted for easier readability. There are some
methods that are no longer used.

Extract the methods and remove unused methods.

Let's try to abstract out some more methods to prevent
arrow-head shaped codes.

jskimdev committed:
Refactor the code

There were several long methods and some variable
naming did not follow the coding standard. Furthermore,
all the methods were public, although they are only used
within the class.

Extract some methods for better readability, change
variable names and change some methods to private.

jskimdev committed:
Merge pull request #2 from jskimdev/branch-A-Assertions


Add assertions to ensure assumptions true

jskimdev committed:
Merge branch 'master' into branch-A-CodeQuality

jskimdev committed:
Merge pull request #3 from jskimdev/branch-A-CodeQuality


Branch a code quality for better readability and efficiency

jskimdev committed:
Remove unused constructor of ByteTalker and method

jskimdev committed:
Add C-Update

jskimdev committed:
Merge pull request #4 from jskimdev/branch-BCD-Extension


Add C-Update

jskimdev committed:
Add product screenshot

jskimdev committed:
Update README.md

jskimdev committed:
Refactor the code and add java doc for public methods and classes

jskimdev committed:
Update README.md

jskimdev committed:
Allow multiple words in find method

jskimdev committed:
Solve a bug in find method

jskimdev committed:
Solve a bug in find method to take multiple words input

jskimdev committed:
Change testcases method names

jskimdev committed:
Prints error message for not providing index in command input

jskimdev committed:
Separate exception classes and handle wrong format error

jskimdev committed:
Solve index error in methods

jskimdev committed:
Refactor import statements

jskimdev committed:
Solve index error in update method

jskimdev committed:
Update README file to fix some errors

jskimdev committed:
Update README to fix some erros

jskimdev committed:
Update README to put input in code block

jskimdev committed:
Refactor long methods
